miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/workflow/factory/BarcodableProgressStepFactory.java

```diff
@@ -0,0 +1,65 @@
+package uk.ac.bbsrc.tgac.miso.service.workflow.factory;
+
+import static uk.ac.bbsrc.tgac.miso.core.data.Barcodable.EntityType;
+import static uk.ac.bbsrc.tgac.miso.core.data.workflow.ProgressStep.FactoryType.BARCODABLE;
+import static uk.ac.bbsrc.tgac.miso.core.data.workflow.ProgressStep.InputType;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import sun.reflect.generics.reflectiveObjects.NotImplementedException;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Use classes from the Java API instead of Sun classes. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1191)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unused import 'sun.reflect.generics.reflectiveObjects.NotImplementedException'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)